src/add/operator/buffer.ts

```diff
@@ -4,4 +4,8 @@ import {buffer} from '../../operator/buffer';
 
 Observable.prototype.buffer = buffer;
 
-export var _void: void;
\ No newline at end of file
+declare module '../../Observable' {
+  interface Observable<T> {
+    buffer: (closingNotifier: Observable<any>) => Observable<T[]>;
```

Comment:
@kwonoj From what I understand, using `buffer: typeof buffer` is not equivalent to using `buffer: (closingNotifier: Observable<any>) => Observable<T[]>;` or `buffer: BufferSignature<T>;`. If we go with `buffer: typeof buffer`, information about `T`, or the type of items that the source produces, is lost. I put together an image to illustrate:

![left is with the current signature (or `interface<T>`), right is with typeof](http://i.imgur.com/HYUmmHG.png)

Left is with the current signature (or `interface<T>`), right is with typeof